src/library/Connect/WorkloadInformationProvider.cs

```diff
@@ -117,6 +122,25 @@ internal class WorkloadInformationProvider : IWorkloadInformationProvider
             }
         }
 
+        public Dictionary<String, IList<int>> GetPortsToIgnoreFromAnnotations(V1ServiceList serviceNameSpace) {
+            Dictionary<String, IList<int>> servicePortsToIgnore = new Dictionary<String, IList<int>>();
+            List<V1Service> servicesWithIgnorePorts = serviceNameSpace.Items.Where(item => item.Metadata?.Annotations?.ContainsKey(DeploymentConfig.ServiceAnnotations) ?? false).ToList();
+            servicesWithIgnorePorts.ForEach(service => {
+                if (service.Metadata?.Annotations?.TryGetValue(DeploymentConfig.ServiceAnnotations, out string ports) ?? false) {
+                    if (ports.Length > 0) {

```

Comment:
💡 We have the value of `ports` above, why are we waiting until now - we could do better error handling by failing early scenarios like:

```.cs
if (ports.Length === 0) {  // error handle stuff here }
```

Reply:
The try catch is for handling any parsing errors.
ports.length == 0 is a valid condition.